### ISSUE TITLE
Tree View scrolls on mouse click with Auto reveal, Pending pane items and Add new tabs at end

### ISSUE THREAD

**Issue by Ben3eeE** [atom/tree-view#867]
### Prerequisites
- [X] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
- [X] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
- [X] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
- [X] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [X] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [X] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?
### Description

When Auto Reveal, Pending Pane Items and Add New Tabs At End is on and with a specific set of steps the `tree-view` scrolls when clicking on it.
I can't reproduce this with either Auto reveal, Pending pane items or Add new tabs at end off.

Discovered in https://github.com/atom/tree-view/issues/844 but since that was closed unresolved I open this because I was able to reproduce it/similar in safe mode.
### Steps to Reproduce
1. Open Atom in safe mode, `atom --safe`.
2. Ensure you have the `tree-view` setting Auto Reveal on, the core setting Pending Pane Items on and the `tabs` setting Add New Tabs at end on.
3. Open a project folder, I used the `image-view` repository.
4. Expand some folders and resize the Atom window so you have scroll bars visible in the `tree-view`.
5. Scroll to the bottom of the `tree-view` and open a file in a permanent tab by double clicking it, I opened `coffeelint.json`.
6. Open another file in a pending tab. I opened `image-editor-view-spec.coffee`.
7. Scroll the tree-view up so the first file, `coffeelint.json`, is no longer visible but `image-editor-view-spec.coffee` is visible.
8. Double click another file, I double clicked `main.coffee`, `tree-view` scrolls to `image-editor-view.coffee`on first click that file opens in a permanent tab on the second click.

Long steps so I include a gif of repro:
![tree view double click2](https://cloud.githubusercontent.com/assets/1058982/16023169/d2bcd962-31bb-11e6-9b89-164043e3beaf.gif)

Also if you don't double click in the 8th step the `tree-view` keeps scrolling every single click + reproduced in another project folder:
![tree view scroll on click](https://cloud.githubusercontent.com/assets/1058982/16023197/f655c6b8-31bb-11e6-8799-96480aca852c.gif)

These steps start from an empty Atom window and use a public repository which is nice. But this is reproducible by just opening some permanent tab then scrolling it out of view and opening some pending tabs.

**Expected behavior:** I expect `tree-view` to not scroll when I am clicking on it with the mouse regardless of various settings/states. I also expect the click events to fire on the correct items? Or I don't know what is going on. In my gif I click on `main.coffee` which should open that in a pending tab and then the `tree-view` scrolls to `image-editor-view.coffee` which should now be the pending tab on the next click.

**Actual behavior:** The `tree-view` scrolls when clicking on it with the mouse. `image-editor-view.coffee` opens in a permanent tab when it is only clicked once.
### Versions

Tested both on Windows 7 and Windows 10, version commands below are from Windows 10 machine.
##### atom --version

```
Atom    : 1.8.0
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```
##### apm --version

```
apm  1.9.2
npm  2.13.3
node 0.10.40
python 2.7.11
git 2.8.1.windows.1
visual studio
```

**Comment by Ben3eeE** [comment]
I just noticed that I also need to have `Add new tabs at end` on to reproduce this.

**Comment by anderoonies** [comment]
this is an issue with tabs, i'm making the PR there

**Comment by jackocnr** [comment]
I'm having exactly the same problem, with `Add new tabs at end` off. In fact I've disabled all packages except settings-view and tree-view, and started in safe mode, and I still get the problem.

I don't see the problem after a fresh start of atom, only after I have scrolled in the tree view. Usually I get it exactly like in the op's second video, where every time i click a file in the tree-view, that file scrolls to the center. Sometimes, after some more scrolling and clicking, I get stuck in another situation like the first video, where if I click a file and then scroll a bit, and then click again, the tree-view undoes the scroll I just did. Attaching another video in case it's useful (apologies - it seems to have gotten slowed down during conversion to gif):

![scroll2](https://cloud.githubusercontent.com/assets/1186883/25442529/347dff66-2a9d-11e7-9e3f-5c26fa1a7cca.gif)

`atom --version`
Atom    : 1.16.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0

`apm --version`
apm  1.16.1
npm  3.10.5
node 6.9.5 x64
python 2.7.13
git 2.11.0

MacOS 10.12.4

Let me know if there's any more testing I can do to help!
